### DISCUSSION TITLE
How to build a docker image for milvus

### DISCUSSION THREAD

**Discussion by Reidddddd** [milvus-io/milvus#32364] [Q&A and General discussion]
My requirement is to publish self-development milvus's docker image to company internal image hub.

If I miss any documents, your pointer is highly appreciated!

**Comment by xiaofan-luan** [comment]
I'm assuming you know docker.
here is the script and docker files we are using https://github.com/milvus-io/milvus/tree/master/build.
Sorry we don't have too much document about it.

**Reply by xiaofan-luan** [reply]
if you hit any issue, @zwd1208 could probably help on that

**Reply by Reidddddd** [reply]
seems to be `build_image.sh` is the one i want, any command needs to be executed before executing `build_image.sh`? ping @zwd1208 as well

**Comment by Reidddddd** [comment]
When execute `build/builder.sh`:

![Screenshot 2024-04-17 at 18 12 41](https://github.com/milvus-io/milvus/assets/5352837/19481525-983f-486e-9a62-67af95d0140d)

text version:
```
prometheus-cpp/1.1.0: WARN: Build folder is dirty, removing it: /home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/build/fa20d22332d4ea571be891e8b4ebf8b917e5e8cd
prometheus-cpp/1.1.0: Copying sources to build folder
ERROR: [('/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/source/src/cmake/project-import-pkgconfig/sample_client.cc', '/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/build/fa20d22332d4ea571be891e8b4ebf8b917e5e8cd/src/cmake/project-import-pkgconfig/sample_client.cc', '[Errno 2] No such file or directory'), ('/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/source/src/cmake/project-import-pkgconfig/sample_server.cc', '/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/build/fa20d22332d4ea571be891e8b4ebf8b917e5e8cd/src/cmake/project-import-pkgconfig/sample_server.cc', '[Errno 2] No such file or directory'), ('/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/source/src/cmake/project-import-cmake/sample_client.cc', '/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/build/fa20d22332d4ea571be891e8b4ebf8b917e5e8cd/src/cmake/project-import-cmake/sample_client.cc', '[Errno 2] No such file or directory'), ('/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/source/src/cmake/project-import-cmake/sample_server.cc', '/home/milvus/.conan/data/prometheus-cpp/1.1.0/_/_/build/fa20d22332d4ea571be891e8b4ebf8b917e5e8cd/src/cmake/project-import-cmake/sample_server.cc', '[Errno 2] No such file or directory')]
Error copying sources to build folder
conan install failed
make: *** [Makefile:212: build-3rdparty] Error 1
```

**Comment by xiaofan-luan** [comment]
@samhuang-z 
could you help on it?

**Comment by yellow-shine** [comment]
@Reidddddd in your case, your conan workspace is dirty. kill the victim, prometheus-cpp
```
rm -rf /home/milvus/.conan/data/prometheus-cpp
```
then, try again

**Reply by Reidddddd** [reply]
Hi, it doesn't work. 

But i found the exact path is `/Users/reid.chen/.conan/data/prometheus-cpp/` not `/home/milvus`

Don't understand why the build tried to find under `/home/milvus`.. anything I can change to make it find the correct path?

**Reply by Reidddddd** [reply]
btw, i think `/Users/reid.chen/.conan/data/prometheus-cpp/` was generated by `make`

**Comment by xiaofan-luan** [comment]
it's generated by conan.
Did you specify a special conan repo?

**Reply by Reidddddd** [reply]
didn't specify

**Comment by xiaofan-luan** [comment]
are u runing on ARM or some special cpu architecture or operating system?

**Reply by Reidddddd** [reply]
macOS, x86_64

**Reply by xiaofan-luan** [reply]
I have the same device and it works fine. It seems that the promethues you download lack of certain files.

try conan remove prometheus-cpp and try to reinstall see if you get luck
